drivers/pinmux/stm32/pinmux_stm32.c

```diff
@@ -52,15 +44,12 @@ const struct device *gpio_ports[STM32_PORTS_MAX] = {
 
 static int stm32_pin_configure(uint32_t pin, uint32_t func, uint32_t altf)
 {
-	/* determine IO port registers location */
-	uint32_t offset = STM32_PORT(pin) * GPIO_REG_SIZE;
-	uint8_t *port_base = (uint8_t *)(GPIO_PORTS_BASE + offset);
+	const struct device *port_device = gpio_ports[STM32_PORT(pin)];
```

Comment:
`STM32_PORT(pin)` bound check may be useful here (not sure if done previous to the call or if needed at all)

Reply:
Right, this is indeed needed